dali/imgcodec/parsers/tiff_test.cc

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+#include <string>
+
+#include "dali/imgcodec/image_source.h"
+#include "dali/imgcodec/parsers/tiff.h"
+#include "dali/test/dali_test.h"
+#include "dali/test/dali_test_config.h"
+
+namespace dali {
+namespace imgcodec {
+namespace test {
+
+class TiffParserTest : public ::testing::Test {
+ public:
+  TiffParserTest() : parser_() {
+    auto filename = testing::dali_extra_path() + "/db/single/tiff/0/cat-1245673_640.tiff";
+    img_ = ImageSource::FromFilename(filename);
+    auto stream = img_.Open();
+    valid_tiff_.resize(stream->Size());
+    size_t offset = 0;
+    size_t n;
+    while ((n = stream->Read(valid_tiff_.data() + offset, 4096))) {
+      offset += n;
+    }
+  }
+
+  bool CanParse(std::vector<char> data) {
+    auto src = ImageSource::FromHostMem(data.data(), data.size());
+    return parser_.CanParse(&src);
+  }
+
+  std::vector<char> valid_tiff_;
+  TiffParser parser_;
+  ImageSource img_;
```

Comment:
Nitpick: can you move the sources (ImgageSource and a vector) to adjacent lines?